wellcomecollection/wellcomecollection.org
# Change GuideStop rendering based on if inside slice simulator or not

## What does this change?

[Slack conversation](https://wellcome.slack.com/archives/CUA669WHH/p1729784277499519)

I noticed that the slice previews were returning 404. So I ran the slice simulator locally and implemented this solution that I'm open to feedback on.

![image](https://github.com/user-attachments/assets/dcd8cea7-7c9c-4ac3-97ce-bcc2fa9a439b)


The main issue was with the `linkResolver` being called, as the Slice Simulator doesn't fetch outside information, it didn't know what to do with the related `exhibitionGuide`'s `uid`. So I added a condition to stop it from using that function. It now rendered:

<img width="271" alt="Screenshot 2024-10-28 at 11 27 03" src="https://github.com/user-attachments/assets/f426104a-24bc-460f-8ecb-fa61ba6a6a19">

> [!WARNING]
> **EDIT: Have discussed everything below with Gareth and removed the code changes. [A new ticket has been created](https://github.com/wellcomecollection/wellcomecollection.org/issues/11342) to address this differently.**

I then thought it was so small that it wasn't that useful, so I implemented this but happy to remove it if we don't want Slice Simulator related things in our components - although I feel like other slices could benefit from a different styling based on where they are rendered.
So, in `GuideStop`, we look at if it is rendered inside a slice simulator iframe or not. If so, we don't put it in a grid, making it easier to visualise. I still wish it could be _better_, but we can't really control the size of the window, sadly, otherwise rendering it in mobile would be much better. Ideas?

After
<img width="267" alt="Screenshot 2024-10-28 at 11 51 14" src="https://github.com/user-attachments/assets/bc7ed14a-4fce-4fcb-86fb-6de532571c4c">


## How to test

Run the slice simulator locally as well as the website. Go to Prismic staging and change the live preview setting link to be http://localhost:3000/slice-simulator (don't forget to switch it back!)

<img width="275" alt="Screenshot 2024-10-28 at 11 57 26" src="https://github.com/user-attachments/assets/d620672b-6205-49d7-9501-725d80213444">

If run the simulator or look at a Prismic document using the guide stop, it should render as described above.



## How can we measure success?

More useful for editors

## Have we considered potential risks?
N/A?
